recipes-core/uboot-env/uboot-env_0.1.bb

```diff
@@ -0,0 +1,27 @@
+#
+# Copyright (c) Siemens AG, 2021
+#
+# Authors:
+#  Quirin Gylstorff <quirin.gylstorff@siemens.com>
+#
+# SPDX-License-Identifier: MIT
+
+inherit dpkg-raw
+
+DESCRIPTION = "prepare u-boot environment for swupdate"
+
+SRC_URI += "file://postinst \
+            file://boot.config \
+            file://update_ustate.sh \
+            file://write-u-boot-env.sh \
+            file://write-u-boot-env.service"
+
+do_install () {
+  install -v -d ${D}/usr/share/u-boot-env
+  install -v -m 640 ${WORKDIR}/boot.config ${D}/usr/share/u-boot-env/boot.config
+  install -v -m 755 ${WORKDIR}/update_ustate.sh ${D}/usr/share/u-boot-env/update_ustate.sh
+  install -v -m 755 ${WORKDIR}/write-u-boot-env.sh ${D}/usr/share/u-boot-env/write-u-boot-env.sh
+
+  install -v -d ${D}/usr/lib/systemd/system
+  install -v -m 666 ${WORKDIR}/write-u-boot-env.service ${D}/usr/lib/systemd/system/write-u-boot-env.service
+}
```

Comment:
`recipes-bsp/patch-u-boot-env/...` or so. It should only do what is currently needed for the legacy firmware on the deployed devices.

We will have to make sure that future firmware will already contain the required logic.

Reply:
I will move it